prometheus_client/metrics.py

```diff
@@ -390,10 +390,15 @@ def dec(self, amount: float = 1) -> None:
         self._raise_if_not_observable()
         self._value.inc(-amount)
 
-    def set(self, value: float) -> None:
-        """Set gauge to the given value."""
+    def set(self, value: float, timestamp_sec: Optional[float] = None) -> None:
```

Comment:
🤔  I wonder if it would be best to start by not taking an additional timestamp_sec variable, and calculate the current time ourselves iff multiprocess_mode is `mostrecent` or `livemostrecent`?

To my knowledge most client libraries avoid letting users easily set timestamps on metrics as it becomes dangerous very quickly. This would also mean that users could use the new functionality without having to manually pass timestamps around.

Reply:
Hmm, I think I would prefer to use the same Gauge, mostly because `TimestampedGauge` sounds like it would export the timestamp as well, which I don't want to include as part of this PR. For inc/dec I think we can just throw if they are used with `mostrecent` or `livemostrecent`. I could be convinced though.